samplers/README.md

```diff
@@ -1,7 +1,65 @@
 # Samplers
 
+## Declarative configuration
+
+The following samplers support [declarative configuration](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/configuration#declarative-configuration):
+
+* `RuleBasedRoutingSampler`
+
+To use:
+
+* Add a dependency on `io.opentelemetry:opentelemetry-sdk-extension-incubator:<version>`
+* Follow the [instructions](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/incubator/README.md#file-configuration) to configure OpenTelemetry with declarative configuration.
+* Configure the `.tracer_provider.sampler` to include the `rule_based_routing` sampler.
```

Comment:
TODO: add usage instructions for otel java agent

Reply:
Doesn't work with the agent yet:
- https://github.com/open-telemetry/opentelemetry-java/pull/6725
- https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12265

For now, add a note in the README that agent support is WIP.